extensions/broadcast_channel/lib.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+mod in_memory_broadcast_channel;
+
+pub use in_memory_broadcast_channel::InMemoryBroadcastChannel;
+
+use async_trait::async_trait;
+use deno_core::error::bad_resource_id;
+use deno_core::error::AnyError;
+use deno_core::include_js_files;
+use deno_core::op_async;
+use deno_core::op_sync;
+use deno_core::Extension;
+use deno_core::OpState;
+use deno_core::Resource;
+use deno_core::ResourceId;
+use deno_core::ZeroCopyBuf;
+use std::cell::RefCell;
+use std::path::PathBuf;
+use std::rc::Rc;
+
+#[async_trait]
+pub trait BroadcastChannel: Clone {
+  type Resource: Resource;
+
+  fn subscribe(&self) -> Result<Self::Resource, AnyError>;
+
+  fn unsubscribe(&self, resource: &Self::Resource) -> Result<(), AnyError>;
+
+  async fn send(
+    &self,
+    resource: &Self::Resource,
+    name: String,
+    data: Vec<u8>,
+  ) -> Result<(), AnyError>;
+
+  async fn recv(
+    &self,
+    resource: &Self::Resource,
+  ) -> Result<Option<Message>, AnyError>;
+}
+
+pub type Message = (String, Vec<u8>);
+
+pub fn op_broadcast_subscribe<BC: BroadcastChannel + 'static>(
+  state: &mut OpState,
+  _args: (),
+  _buf: Option<ZeroCopyBuf>,
+) -> Result<ResourceId, AnyError> {
+  let bc = state.borrow::<BC>();
+  let resource = bc.subscribe()?;
+  Ok(state.resource_table.add(resource))
+}
+
+pub fn op_broadcast_unsubscribe<BC: BroadcastChannel + 'static>(
+  state: &mut OpState,
+  rid: ResourceId,
+  _buf: Option<ZeroCopyBuf>,
```

Comment:
I think the new op signatures should allow for this:
```suggestion
  _: (),
```